webapp/src/ts/services/create-user-for-contacts.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Injectable } from '@angular/core';
+import { DbService } from '@mm-services/db.service';
+import { DBSyncService, SyncStatus } from '@mm-services/db-sync.service';
+import { SessionService } from '@mm-services/session.service';
+import { SettingsService } from '@mm-services/settings.service';
+import { UserSettingsService } from '@mm-services/user-settings.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CreateUserForContactsService {
+  constructor(
+    private settingsService: SettingsService,
+    private userSettingsService: UserSettingsService,
+    private dbService: DbService,
+    private dbSyncService: DBSyncService,
+    private sessionService: SessionService,
+  ) {
+    const that = this;
+    this.settingsService
+      .get()
+      .then((settings) => {
+        if (settings?.transitions?.create_user_for_contacts) {
+          that.dbSyncService.subscribe(async(status) => {
+            return that.syncStatusChanged(status);
```

Comment:
I swear at one point I needed the `bind/that` hack. But now that I try it again, it seems to work fine without either.  I believe it is because of my arrow function, but I was 99% sure I was using the arrow function before to...  Oh well. Works now!  :+1:

Reply:
My gut feeling is `bind` will be necessary, since the function will run in a different context later - I'm referring to the callback that you pass to subscribe.